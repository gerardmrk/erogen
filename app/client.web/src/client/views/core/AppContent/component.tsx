import * as React from "react";
import styles from "./styles.scss";
import { LocalProps, StoreProps, DispatchProps } from ".";
import { Switch } from "react-router";
import LoadingDisplay from "./LoadingDisplay";
import routeConfs from "@client/views/conf.routes";
import EnhancedRoute from "@client/views/components/EnhancedRoute";

type Props = LocalProps & StoreProps & DispatchProps;

type State = {};

export class AppContent extends React.PureComponent<Props, State> {
  public render() {
    return (
      <main className={styles.main}>
        <React.Suspense fallback={<LoadingDisplay />}>
          <Switch>
            {routeConfs.map((r, i) => (
              <EnhancedRoute key={i} {...r} />
            ))}
          </Switch>
        </React.Suspense>
      </main>
    );
  }
}

export default AppContent;
