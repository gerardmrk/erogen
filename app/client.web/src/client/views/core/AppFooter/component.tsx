import * as React from "react";
import { Pane } from "evergreen-ui";
import styles from "./styles.scss";
import { LocalProps, StoreProps, DispatchProps } from ".";
import Container from "@client/views/components/Container";

type Props = LocalProps & StoreProps & DispatchProps;

type State = {};

export class AppFooter extends React.PureComponent<Props, State> {
  public render() {
    return (
      <Pane className={styles.main} border={"muted"} background={"tint3"}>
        <Container className={styles.container}>{"AppFooter"}</Container>
      </Pane>
    );
  }
}

export default AppFooter;
