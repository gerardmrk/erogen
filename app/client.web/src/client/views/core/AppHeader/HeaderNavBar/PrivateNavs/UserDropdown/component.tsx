import * as React from "react";
import { LocalProps, StoreProps, DispatchProps } from ".";
import Dropdown from "@client/views/components/ui.modules/Dropdown";

type Props = LocalProps & StoreProps & DispatchProps;

type State = {};

export class UserDropdown extends React.PureComponent<Props, State> {
  private onProfileNavClick = () => {
    if (this.props.profile) {
      this.props.history.push(`/user/${this.props.profile.username}`);
    }
  };

  private onDashboardNavClick = () => {
    this.props.history.push("/dashboard");
  };

  private onSettingsNavClick = () => {
    this.props.history.push("/settings");
  };

  private onLogoutActionClick = () => {
    this.props.logout();
  };

  public render() {
    return (
      <Dropdown icon={"user circle"} direction={"left"}>
        <Dropdown.Menu direction={"left"}>
          <Dropdown.Header>
            {this.props.profile ? this.props.profile.username : "user"}
          </Dropdown.Header>

          <Dropdown.Divider />

          <Dropdown.Item onClick={this.onProfileNavClick}>
            {"Profile"}
          </Dropdown.Item>

          <Dropdown.Item onClick={this.onDashboardNavClick}>
            {"Dashboard"}
          </Dropdown.Item>

          <Dropdown.Divider />

          <Dropdown.Item onClick={this.onSettingsNavClick}>
            {"Settings"}
          </Dropdown.Item>

          <Dropdown.Item onClick={this.onLogoutActionClick}>
            {"Logout"}
          </Dropdown.Item>
        </Dropdown.Menu>
      </Dropdown>
    );
  }
}
