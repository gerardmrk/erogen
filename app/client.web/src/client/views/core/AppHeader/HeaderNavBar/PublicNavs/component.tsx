import * as React from "react";
import styles from "./component.styles.scss";
import { LocalProps, StoreProps, DispatchProps } from ".";
import NavLink from "@client/views/components/NavLink";
import { headerNavsPublic } from "@client/views/conf.navs";

type Props = LocalProps & StoreProps & DispatchProps;

type State = {};

export class PublicNavs extends React.PureComponent<Props, State> {
  public render() {
    return (
      <nav className={styles.main}>
        <div className={styles.otherNavsGroup}>
          {headerNavsPublic.map((n, i) => (
            <div className={styles.navBox} key={i}>
              <NavLink path={n.to} label={n.label} />
            </div>
          ))}
        </div>

        <div className={styles.defaultNavsGroup}>
          <div className={styles.navBox}>
            <NavLink
              path={"/login"}
              label={this.props.t("navs.public.login")}
            />
          </div>

          <div className={styles.navBox}>{"/"}</div>

          <div className={styles.navBox}>
            <NavLink
              path={"/register"}
              label={this.props.t("navs.public.register")}
            />
          </div>
        </div>
      </nav>
    );
  }
}
