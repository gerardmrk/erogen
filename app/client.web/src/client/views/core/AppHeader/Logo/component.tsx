import * as React from "react";
import styles from "./component.styles.scss";
import logoSVG from "@client/logo.svg";
import { Link } from "react-router-dom";
import { DEFAULT_PUBLIC_PATH } from "@client/views/conf.routes";

export class Logo extends React.PureComponent {
  public render() {
    return (
      <div className={styles.main}>
        <Link to={DEFAULT_PUBLIC_PATH}>
          <img src={logoSVG} alt={"brand logo"} />
        </Link>
      </div>
    );
  }
}

export default Logo;
