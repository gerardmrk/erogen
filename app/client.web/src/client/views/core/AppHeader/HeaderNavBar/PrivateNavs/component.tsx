import * as React from "react";
import styles from "./component.styles.scss";
import { LocalProps, StoreProps, DispatchProps } from ".";
import UserDropdown from "./UserDropdown";
// import Menu from "@client/views/components/ui.collections/Menu";

type Props = LocalProps & StoreProps & DispatchProps;

type State = {};

export class PrivateNavs extends React.PureComponent<Props, State> {
  public render() {
    return (
      <div className={styles.main}>
        <div className={styles.navs}>{"..."}</div>
        <div className={styles.userdropdown}>
          <UserDropdown />
        </div>
      </div>
    );
  }
}
