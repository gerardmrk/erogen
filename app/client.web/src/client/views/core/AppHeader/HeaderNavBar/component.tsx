import * as React from "react";
import styles from "./styles.scss";
import { LocalProps, StoreProps, DispatchProps } from ".";
import PrivateNavs from "./PrivateNavs";
import PublicNavs from "./PublicNavs";

export type Props = LocalProps & StoreProps & DispatchProps;
export type State = {};

export class HeaderNavBar extends React.Component<Props, State> {
  public render() {
    return (
      <div className={styles.main}>
        {this.props.isAuthenticated ? <PrivateNavs /> : <PublicNavs />}
      </div>
    );
  }
}

export default HeaderNavBar;
