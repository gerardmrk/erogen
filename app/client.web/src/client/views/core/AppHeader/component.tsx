import * as React from "react";
import styles from "./styles.scss";
import { Pane } from "evergreen-ui/esm/layers";
import { LocalProps, StoreProps, DispatchProps } from ".";
import Container from "@client/views/components/Container";
import Logo from "./Logo";
import HeaderNavBar from "./HeaderNavBar";

export type Props = LocalProps & StoreProps & DispatchProps;

export type State = {};

export class AppHeader extends React.PureComponent<Props, State> {
  public render() {
    return (
      <Pane
        borderColor={"x"}
        is={"header"}
        border={"muted"}
        background={"blueTint"}
      >
        <Container className={styles.container}>
          <div className={styles.logo}>
            <Logo />
          </div>
          <HeaderNavBar />
        </Container>
      </Pane>
    );
  }
}

export default AppHeader;
