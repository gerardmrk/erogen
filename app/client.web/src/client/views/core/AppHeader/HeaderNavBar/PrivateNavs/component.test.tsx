import * as React from "react";
import { create, ReactTestRenderer, ReactTestInstance } from "react-test-renderer"; // prettier-ignore

import { PrivateNavs } from "./component";

const Component = PrivateNavs;

describe("<PrivateNavs/>", () => {
  let renderer: ReactTestRenderer;
  let component: ReactTestInstance;

  beforeEach(() => {
    renderer = create(<Component />);
    component = renderer.getInstance() as ReactTestInstance;
  });

  it("renders ok", () => {
    expect(component).toBeDefined();
  });
});
