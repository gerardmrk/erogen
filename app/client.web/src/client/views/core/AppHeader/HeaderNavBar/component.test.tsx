// Only enable snapshotting when component has stabilised and is not
// expected to undergo much development changes in the future.
const SNAPSHOT_ENABLED = true;

import * as React from "react";
import { ReactTestRenderer, ReactTestInstance } from "react-test-renderer"; // prettier-ignore
import { createRenderer } from "@client/views/_fixtures_/create-renderer";

import { HeaderNavBar } from "./component";

const Component = HeaderNavBar;

describe("<HeaderNavBar/>", () => {
  let renderer: ReactTestRenderer;
  let component: ReactTestInstance;

  beforeEach(() => {
    renderer = createRenderer(<Component isAuthenticated={true} />);
    component = renderer.getInstance() as ReactTestInstance;
  });

  if (SNAPSHOT_ENABLED) {
    test("snapshot", () => {
      expect(renderer.toJSON()).toMatchSnapshot();
    });
  }

  it("renders ok", () => {
    expect(component).toBeDefined();
  });
});
