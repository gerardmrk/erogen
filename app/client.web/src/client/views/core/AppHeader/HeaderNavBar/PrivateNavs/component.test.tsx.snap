// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PrivateNavs/> snapshot 1`] = `
<div
  className="main"
>
  <div
    className="navs"
  >
    navs
  </div>
  <div
    className="userdropdown"
  >
    <div
      aria-expanded={false}
      className="ui dropdown"
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      role="listbox"
      tabIndex={0}
    >
      <div
        aria-live="polite"
        className="text"
        role="alert"
      />
      <i
        aria-hidden="true"
        className="user circle icon"
        onClick={[Function]}
      />
      <div
        className="left menu transition left"
      >
        <div
          className="header"
        >
          xoxo
        </div>
        <div
          className="divider"
        />
        <div
          className="item"
          onClick={[Function]}
          role="option"
        >
          Profile
        </div>
        <div
          className="item"
          onClick={[Function]}
          role="option"
        >
          Dashboard
        </div>
        <div
          className="divider"
        />
        <div
          className="item"
          onClick={[Function]}
          role="option"
        >
          Settings
        </div>
        <div
          className="item"
          onClick={[Function]}
          role="option"
        >
          Logout
        </div>
      </div>
    </div>
  </div>
</div>
`;
